lib/datasource/go/index.ts

```diff
@@ -108,18 +108,19 @@ async function getDatasource(goModule: string): Promise<DataSource | null> {
  *  - Call the respective getReleases in github/gitlab to retrieve the tags
  *  - Filter module tags according to the module path
  */
-export async function getReleases({
-  lookupName,
-}: GetReleasesConfig): Promise<ReleaseResult | null> {
+export async function getReleases(
+  config: GetReleasesConfig
+): Promise<ReleaseResult | null> {
+  const { lookupName } = config;
   logger.trace(`go.getReleases(${lookupName})`);
   const source = await getDatasource(lookupName);
   if (source?.datasource !== github.id && source?.datasource !== gitlab.id) {
     return null;
   }
   const res =
     source.datasource === github.id
-      ? await github.getReleases(source)
-      : await gitlab.getReleases(source);
+      ? await github.getReleases({ ...config, ...source })
+      : await gitlab.getReleases({ ...config, ...source });
```

Comment:
Why this change?

Reply:
This is the only place where the global config wasn't actually passed down to a datasource (according to TypeScript) - so this code was wrong to begin with.

Again, if you want me to, I'll change it to have `ReleasesConfigBase` just have `trustLevel?: "low" | "high"`, but again, that's not what the renovate codebase effectively does.

Reply:
@rarkins @JamieMagee I like the interface change and the go datasource change, but i think we should move it to a separate pr.

Reply:
Makes sense 👍 I did the "simple thing" you suggested for this PR in the meantime.